backend/code/m-booking/src/test/java/com/mbooking/controller/AuthenticationControllerIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+package com.mbooking.controller;
+
+import com.mbooking.dto.ChangePasswordRequestDTO;
+import com.mbooking.dto.LoginRequestDTO;
+import com.mbooking.dto.UserDTO;
+import com.mbooking.exception.ApiAuthException;
+import com.mbooking.model.Customer;
+import com.mbooking.repository.AuthorityRepository;
+import com.mbooking.repository.CustomerRepository;
+import com.mbooking.utils.DatabaseHelper;
+import com.mbooking.utils.SecurityHelper;
```

Comment:
This might be unneccessery for "@Secured" controllers

In WebSecurityConfig add .httpBasic(); on .addFilterBefore()

and use 

ResponseEntity<Type> result = restTemplate.withBasicAuth(username, password)
				.postForEntity(url, dto, type);

Reply:
You misspelled unnecessary